<!--REQUEST_ID=966c273a-ff9a-4ea9-841b-3ddcd379ac4d-->
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49631&attempt_id=1a6fb5de-1777-49cb-a519-224d1366bd31)

Attempted: oidentd

<details><summary>Partial log (click to expand)</summary><p>

```
checking for unistd.h... yes
checking for pid_t... yes
checking for size_t... yes
checking whether time.h and sys/time.h may both be included... yes
checking whether struct tm is in sys/time.h or time.h... time.h
checking build system type... aarch64-unknown-linux-gnu
checking host system type... aarch64-unknown-linux-gnu
configure: error: No lexer generator found; please install flex
builder for '/nix/store/f8nb2hgxw9whqh6zbq1s31vcazvxh6mx-oidentd-2.3.1.drv' failed with exit code 1
error: build of '/nix/store/f8nb2hgxw9whqh6zbq1s31vcazvxh6mx-oidentd-2.3.1.drv' failed
```
</p></details>

--- a/nixos/modules/services/networking/oidentd.nix
+++ b/nixos/modules/services/networking/oidentd.nix
@@ -28,8 +28,7 @@ with lib;
       after = [ "network.target" ];
       wantedBy = [ "multi-user.target" ];
       serviceConfig.Type = "forking";
-      script = "${pkgs.oidentd}/sbin/oidentd -u oidentd -g nogroup" +
-          optionalString config.networking.enableIPv6 " -a ::";
+      script = "${pkgs.oidentd}/sbin/oidentd -u oidentd -g nogroup";
     };
 
     users.users.oidentd = {

--- a/pkgs/servers/identd/oidentd/default.nix
+++ b/pkgs/servers/identd/oidentd/default.nix
@@ -1,19 +1,19 @@
-{ stdenv, fetchurl }:
+{ stdenv, fetchurl, bison, flex }:
 
 stdenv.mkDerivation rec {
   name = "oidentd-${version}";
-  version = "2.2.2";
-
-  CFLAGS = [ "--std=gnu89" ];
+  version = "2.3.1";
+  nativeBuildInputs = [ bison flex ];
 
   src = fetchurl {
-    url = "https://ftp.janikrabe.com/pub/oidentd/releases/${version}/${name}.tar.gz";
-    sha256 = "1svj7ymljp4s17d7jlx6602n9081714qsj5yymmv1s9wagzjqyn9";
+    url = "https://files.janikrabe.com/pub/oidentd/releases/${version}/${name}.tar.gz";
+    sha256 = "1sljid4jyz9gjyx8wy3xd6bq4624dxs422nqd3mcxnsvgxr6d6zd";
   };
 
-  meta = {
-    homepage = http://ojnk.sourceforge.net/;
-    description = "An implementation of the IDENT protocol";
-    platforms = stdenv.lib.platforms.linux;
+  meta = with stdenv.lib; {
+    description = "Configurable Ident protocol server";
+    homepage = https://oidentd.janikrabe.com/;
+    license = licenses.gpl2;
+    platforms = platforms.linux;
   };
 }

